connectors/katalog/pkg/connector/connector.go

```diff
@@ -57,3 +71,100 @@ func (r *Handler) getAssetInfo(c *gin.Context) {
 
 	c.JSON(http.StatusOK, &response)
 }
+
+func (r *Handler) reportError(errorMessage string, c *gin.Context, httpCode int) {
+	r.log.Error().Msg(errorMessage)
+	c.JSON(httpCode, gin.H{"error": errorMessage})
+}
+
+// Enables writing of assets to katalog. The different flows supported are:
+// (a) When DestinationAssetID is specified:
+//     Then a destination asset id is created with name : <DestinationAssetID>
+// (b) When DestinationAssetID is not specified but ResourceMetadata.Name of source asset is specified:
+//     Then an asset is created with name: ResourceMetadata.Name-<RANDOMSTRING_LENGTH_4>
+// (c) When DestinationAssetID and ResourceMetadata.Name of source asset are not specified:
+//     Then an asset is created with name: fybrik-asset-<RANDOMSTRING_LENGTH_4>
+func (r *Handler) createAssetInfo(c *gin.Context) {
+	// Parse request
+	var request datacatalog.CreateAssetRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		r.reportError("Error during ShouldBindJSON in createAssetInfo"+err.Error(), c, http.StatusBadRequest)
+		return
+	}
+
+	logging.LogStructure("CreateAssetRequest object received:", request, &r.log, false, false)
+
+	if request.DestinationCatalogID == "" {
+		r.reportError("Invalid DestinationCatalogID in request", c, http.StatusBadRequest)
+		return
+	}
+
+	var asset *v1alpha1.Asset
+	var err error
+	assetNamePrefix := ""
+	namespace := request.DestinationCatalogID
+	if request.DestinationAssetID != "" {
+		asset = &v1alpha1.Asset{
+			ObjectMeta: v1.ObjectMeta{Namespace: namespace, Name: request.DestinationAssetID},
+			Spec: v1alpha1.AssetSpec{
+				SecretRef: v1alpha1.SecretRef{Name: request.Credentials},
+				Metadata:  request.ResourceMetadata,
+				Details:   request.Details,
+			},
+		}
+	} else {
+		if request.ResourceMetadata.Name != "" {
+			assetNamePrefix = request.ResourceMetadata.Name
+		} else {
+			assetNamePrefix = "fybrik-asset"
+		}
+		r.log.Info().Msg("Using assetNamePrefix : " + assetNamePrefix)
+		asset = &v1alpha1.Asset{
+			ObjectMeta: v1.ObjectMeta{Namespace: namespace, GenerateName: assetNamePrefix},
+			Spec: v1alpha1.AssetSpec{
+				SecretRef: v1alpha1.SecretRef{Name: request.Credentials},
+				Metadata:  request.ResourceMetadata,
+				Details:   request.Details,
+			},
+		}
+	}
+	logging.LogStructure("Fybrik Asset to be created in Katalog:", asset, &r.log, false, false)
+
+	err = r.client.Create(context.Background(), asset)
+	if err != nil {
+		r.reportError("Error during create asset. Error:"+err.Error(), c, http.StatusInternalServerError)
+		return
+	}
+	logging.LogStructure("Created Asset: ", asset, &r.log, false, false)
+
+	if assetNamePrefix != "" {
+		asset.Spec.Metadata.Name = asset.ObjectMeta.Name
+		err = r.client.Update(context.Background(), asset)
```

Comment:
you have to check that the error is `Conflict` 
```
if err != nil && errors.IsConflict(err) 
```

Reply:
this call was redundant. Fixed this issue. Please check. Thanks.